onyo/lib/tests/test_git.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+
+import pytest
+from onyo import OnyoRepo, OnyoInvalidRepoError
+from onyo.lib.git import GitRepo
+
+
+def test_GitRepo_instantiation_existing(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly for paths to existing
+    repositories.
+    """
+    new_repo = GitRepo(repo.git.root)
+    assert new_repo.root.samefile(repo.git.root)
+
+
+@pytest.mark.repo_dirs('existing/directory')
+def test_GitRepo_instantiation_find_root(repo: OnyoRepo) -> None:
+    """
+    The GitRepo class must instantiate correctly with paths to sub-directories
+    in a repository when `find_root=True`.
```

Comment:
I'd turn this into a test of the `find_root` parameter, not just one case for it.
So, yes, it should instantiate with `find_root=True`. But it should also fail with `find_root=False`.

Reply:
> But it should also fail with find_root=False.

Indeed it should have been tested that it fails, because it did not. I created the issue #399 for it, and fixed it. I then proceded and added the test for it!

Thanks for the comment!